docs/gpt.rst

```diff
@@ -0,0 +1,67 @@
+gpt
+=======
+
+.. dfhack-tool::
+    :summary: AI-generated written content!
+    :tags: fort gameplay
+
+Enables a UI for submitting knowledge item descriptions to OpenAI for generating
+poetry, star charts, and excerpts from longer works such as biographies, dictionaries,
+treatises on technological evolution, comparative biographies, cultural histories,
+autobiographies, cultural comparisons, essays, guides, manuals, and more.
+
+``enable gpt``
+=======
+Enables the plugin. The overlay will be shown when a knowledge item or unit view sheet is open.
+
+``disable gpt``
+=======
+Disables the plugin.
+
+Setup:
+
+1. Register for an OpenAI API account. It must be a paid or active trial account.
+2. Generate an API token for your account.
+3. Save your OpenAI API token to a file at the root of your DF directory, `oaak.txt`.
+4. Install python. We used version 3.11 installed from the Microsoft Store.
+5. Install python dependencies Flask and OpenAI: `pip install Flask` and `pip install OpenAI`.
+6. Start the local helper python app: cd into dfhack/scripts directory & run `python gptserver.py`.
+
+Once the python helper is running, you may now enable and use the gpt plugin.
+
+The python script defaults to using the fast, cheap, legacy model `text-davinci-003`.
+If you wish to use the slower, more expensive `gpt-3.5-turbo` or `gpt-4` models, you
+can start the script with `python gptserver.py -gpt3` or `python gptserver.py -gpt4`.
+Tweaking additional OpenAI API parameters will require modifying `gptserver.py` to suit
+your particular desires, until such time as someone may have added additional
+configuration options in a future update to DFHack :D
```

Comment:
it would be fairly trivial to have a configuration screen that can open from the overlay and send options data to the python server